hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -942,6 +959,10 @@ protected void rollbackFailedWrites(Map<String, Option<HoodiePendingRollbackInfo
   }
 
   protected void rollbackFailedWrites(Map<String, Option<HoodiePendingRollbackInfo>> instantsToRollback, boolean skipLocking) {
+    rollbackFailedWrites(instantsToRollback, skipLocking, false);
+  }
+
+  protected void rollbackFailedWrites(Map<String, Option<HoodiePendingRollbackInfo>> instantsToRollback, boolean skipLocking, boolean skipVersionCheck) {
```

Comment:
need to create new method for handling two cases . 
1. rollback outside of migration path (should check version)
2.rollback done pre upgrade/downgrade/migration (should be allowed to skip)

This is an open issue, we can come back to after making rollback table version aware.